### DISCUSSION TITLE
Can somebody help me spot where habits are read from the DB?

### DISCUSSION THREAD

**Discussion by vincenzopronesti** [iSoron/uhabits#720] [Technical Discussion]
I'm trying to find out where habits are read from the underlying DB and inserted into the activity that lists all the habits.

**Comment by hiqua** [comment]
Have you looked into `HabitsApplication` and `HabitsModule`? I would start from there.

**Reply by vincenzopronesti** [reply]
Thanks. I think what I'm looking for may be inside `HabitsApplication`.
```
val db = DatabaseUtils.getDatabaseFile(this)
HabitsApplication.component = DaggerHabitsApplicationComponent
            .builder()
            .appContextModule(AppContextModule(context))
            .habitsModule(HabitsModule(db))
            .build()
DateUtils.setStartDayOffset(3, 0)
val habitList = component.habitList
```
From what I see it gets the `db` and does something with Dagger (I'm new to Dagger so I don't really get it). The part that most interests me in this piece of code is the last line. It uses `component` to get a list of all the habits in the DB. Is there a way I can access the same list from the `EmptyListView` class? Can I use Dagger to inject `HabitsApplication` into `EmptyListView` and then get `component` and than the list of habits?

**Reply by hiqua** [reply]
According to https://github.com/iSoron/uhabits/issues/700#issue-777519989 I think you should modify `updateEmptyView` so that `llEmpty` knows what to show.

**Comment by iSoron** [comment]
Habits are currently loaded from the database when we make the first call to any method in `SQLiteHabitList`.

**Comment by hiqua** [comment]
@vincenzopronesti did you make progress on this, or do you need help?
